src/service-listener.js

```diff
@@ -9,8 +9,31 @@ const { updateConsul } = require('./deployment-change-handlers/update-consul');
 const { updateServiceRegistry } = require('./deployment-change-handlers/update-service-registry');
 const { updateVersionService } = require('./deployment-change-handlers/update-version-service');
 
+function isUpdateAboutOldDeployment(deployment) {
+  return deployment.object.metadata.generation !== deployment.object.status.observedGeneration;
+}
+
+function isStatusAboutOngoingDeployment(deployment) {
+  return !deployment.object.status.conditions.some(c => c.reason === 'NewReplicaSetAvailable');
+}
+
+function ignoreUpdateAboutOngoingDeployment(deployment) {
+  if (deployment.type !== 'MODIFIED') {
+    return false;
+  }
+
+  return isUpdateAboutOldDeployment(deployment) || isStatusAboutOngoingDeployment(deployment);
+}
+
 function onDeploymentChange(deploymentObject) {
   logger.trace('Received deployment object: ', JSON.stringify(deploymentObject, null, 2));
+
+  if (ignoreUpdateAboutOngoingDeployment(deploymentObject)) {
```

Comment:
I was considering whether this should assume that all handlers below won't want these filtered events. However, until that need is identified I think we can merge this.